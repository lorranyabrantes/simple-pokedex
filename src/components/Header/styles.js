import { StyleSheet, Platform } from "react-native";
import { COLORS, FONT_SIZES, SPACING } from "../../services/styles";

const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.RED,
    height: Platform.OS === "ios" ? 80 : 60,
    alignItems: "center",
    justifyContent: "center",
    paddingTop: Platform.OS === "ios" ? 20 : 0,
  },
  title: {
    color: COLORS.WHITE,
    fontSize: FONT_SIZES.xlarge,
  },
  button: {
    position: "absolute",
    top: Platform.OS === "ios" ? 40 : SPACING.large,
    left: SPACING.medium,
  },
  icon: { width: 15, height: 15 },
});

export default styles;
