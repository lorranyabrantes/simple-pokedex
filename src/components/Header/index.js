import React from "react";
import { View, TouchableOpacity, Image } from "react-native";

import iconLeftArrow from "../../assets/png/icon-left-arrow.png";

import Text from "../Text";

import styles from "./styles";

const Header = ({ pageTitle, goBack }) => {
  return (
    <View style={styles.container}>
      {goBack ? (
        <TouchableOpacity
          testID="go-back-button"
          style={styles.button}
          onPress={goBack}
        >
          <Image style={styles.icon} source={iconLeftArrow} />
        </TouchableOpacity>
      ) : null}
      <Text style={styles.title}>{pageTitle}</Text>
    </View>
  );
};

export default Header;
