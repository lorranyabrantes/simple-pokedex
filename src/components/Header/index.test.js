import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import Header from "./";

const props = {
  pageTitle: "Page",
  goBack: jest.fn(),
};

describe("Header", () => {
  test("should render correctly", async () => {
    const component = render(<Header {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  test("expected the function to be called", async () => {
    const { getByTestId } = render(<Header {...props} />);

    fireEvent.press(getByTestId("go-back-button"));

    expect(props.goBack).toHaveBeenCalled();
  });
});
