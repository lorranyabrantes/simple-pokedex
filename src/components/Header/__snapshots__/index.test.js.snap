// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EF5350",
      "height": 80,
      "justifyContent": "center",
      "paddingTop": 20,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 16,
        "opacity": 1,
        "position": "absolute",
        "top": 40,
      }
    }
    testID="go-back-button"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/png/icon-left-arrow.png",
        }
      }
      style={
        Object {
          "height": 15,
          "width": 15,
        }
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#333333",
          "fontFamily": "Roboto-Regular",
          "fontSize": 14,
        },
        Object {
          "color": "#FFFFFF",
          "fontSize": 18,
        },
      ]
    }
  >
    Page
  </Text>
</View>
`;
